### PR TITLE
Iris Network lesson corrected

### PR DESCRIPTION
**Pull request by YaroslavMayorov**

### CODE REVIEW

File: Iris Network/Backpropagation/task.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ that determines the step size at each iteration while approaching the minimum of
 </details>
 
 <h3>Task</h3>
-In the `network.py` file, implement the method `backward` of the `NN` class, which performs the following operations:
+In the `network.py` file, implement the method `backward` of the `NN` class only, which performs the following operations:
```
Comment (stephen-hero) [inline Iris Network/Backpropagation/task.md]: In the `network.py` file, implement only the `backward` method of the `NN` class, which performs the following operations:

File: Iris Network/Forward Step/evaluate.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
 def accuracy(nn, X_test, y_test):
+    # feed the test dataset to the predict method
     nn_y = nn.predict(X_test)
-    return ((nn_y > 0.5).astype(int) == y_test).sum() / len(y_test)
+    # compare the resulting class labels with real ones are return the fraction
+    # of correctly classified objects. Everything greater than 0.5 should be considered a 1,
+    # everything less - a 0.
```
Comment (stephen-hero) [inline Iris Network/Forward Step/evaluate.py]: values less than 0.5 should be rounded to 0.

File: Iris Network/Neural Network/evaluate.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
 def accuracy(nn, X_test, y_test):
+    # feed the test dataset to the predict method
     nn_y = nn.predict(X_test)
-    return ((nn_y > 0.5).astype(int) == y_test).sum() / len(y_test)
+    # compare the resulting class labels with real ones are return the fraction
```
Comment (stephen-hero) [inline Iris Network/Neural Network/evaluate.py]: compare the resulting class labels with the real ones and return the fraction

File: Iris Network/Neural Network/network.py [hunk 4]
```diff
@@ -2,25 +2,52 @@
...
 
+    # This method is to be implemented in the task "Backpropagation".
     def backward(self, X, y, output, learning_rate=0.01):
-        l2_delta = (y - output) * sigmoid_derivative(output)
-        l1_delta = np.dot(l2_delta, self.w2.T) * sigmoid_derivative(self.layer1)
-        self.w2 += (np.dot(self.layer1.T, l2_delta) * learning_rate)
-        self.w1 += (np.dot(X.T, l1_delta) * learning_rate)
+        pass
+        # Calculate output layer error as difference of real class labels and output of
+        # the network, and multiply it elementwise by the derivative of the sigmoid
```
Comment (stephen-hero) [inline Iris Network/Neural Network/network.py]: and multiply it element-wise by the sigmoid derivative of

File: Iris Network/Neurons/task.py [hunk 5]
```diff
@@ -25,7 +31,7 @@ def plot_data(X, y):
     plt.show()
 
 
-# This function splits the dataset into train set and test set using a provided ratio for the split
+# This function splits the dataset into train set and test set using a provided ratio for the split.
```
Comment (stephen-hero) [inline Iris Network/Neurons/task.py]: This function splits the dataset into train and test sets based on the provided split ratio.

File: Iris Network/Train and Predict/evaluate.py [hunk 6]
```diff
@@ -1,3 +1,7 @@
 def accuracy(nn, X_test, y_test):
+    # feed the test dataset to the predict method
     nn_y = nn.predict(X_test)
-    return ((nn_y > 0.5).astype(int) == y_test).sum() / len(y_test)
+    # compare the resulting class labels with real ones are return the fraction
```
Comment (stephen-hero) [inline Iris Network/Train and Predict/evaluate.py]: compare the resulting class labels with the real ones and return the fraction

File: Iris Network/Train and Predict/evaluate.py [hunk 7]
```diff
@@ -1,3 +1,7 @@
...
+    # of correctly classified objects. Everything greater than 0.5 should be considered a 1,
+    # everything less - a 0.
```
Comment (stephen-hero) [inline Iris Network/Train and Predict/evaluate.py]: values less than 0.5 should be rounded to 0.

**Review by stephen-hero — CHANGES_REQUESTED** [review]
language checked

File: Iris Network/Backpropagation/task.md [hunk 8]
```diff
@@ -76,9 +76,16 @@ In the `network.py` file, implement the method `backward` of the `NN` class, whi
 </ul>
 
 Before you start, delete the `pass` operator and uncomment all lines that are not task commentaries.
```
[the remark this answers is not in the document]
Comment (YaroslavMayorov) [reply Iris Network/Backpropagation/task.md]: This one: delta_l2 = Calculate the error for the output layer
delta_l1 = Calculate the error for the hidden layer
self.w2 += Update the weight coefficients of the output layer
self.w1 += Update the weight coefficients of the hidden layer
